src/oss-tests/PackageExistenceTests.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. Licensed under the MIT License.
+
+namespace Microsoft.CST.OpenSource.Tests;
+
+using Contracts;
+using FluentAssertions;
+using Model.Enums;
+using Model.PackageExistence;
+using System.Collections.Generic;
+using VisualStudio.TestTools.UnitTesting;
+
+[TestClass]
+public class PackageExistenceTests
+{
+    /// <summary>
+    /// PackageRemoved should inherit type PackageNotFound as well.
+    /// </summary>
+    [TestMethod]
+    public void PackageExistence_Removed_Is_NotFound()
+    {
+        IPackageExistence packageExistenceRemoved = new PackageRemoved(new HashSet<PackageRemovalReason>(new[]
+        {
+            PackageRemovalReason.PackageUnpublished
+        }));
+
+        packageExistenceRemoved.Should().BeAssignableTo<PackageNotFound>()
+            .Subject.HasEverExisted.Should().BeTrue();
+
+        packageExistenceRemoved.HasEverExisted.Should().BeTrue();
```

Comment:
Small nit: this assertion appears redundant with the one on line 27. Looks like the same redundancy appeared in the below test case as well.

Reply:
Yeah I guess it's kinda redundant but I added it for sanity sake. I don't particularly see a problem with having it, but I understand the confusion/redundancy reason to remove it.